Sources/SwiftParser/Lexer/Cursor.swift

```diff
@@ -402,6 +405,15 @@ extension Lexer.Cursor {
     } else {
       newlineInLeadingTrivia = .absent
     }
+    
+    var flags: Lexer.Lexeme.Flags = []
+    if newlineInLeadingTrivia == .present {
+      flags.insert(.isAtStartOfLine)
+    }
+    if let previousLexemeTrailingNewlinePresence, previousLexemeTrailingNewlinePresence == .present {
+      flags.insert(.isAtStartOfLine)
+    }
+    self.previousLexemeTrailingNewlinePresence = nil
```

Comment:
Resetting to `nil` doesn’t really make sense here, right? If we follow the same meaning as for `previousTokenKind`, `nil` means that there hasn’t been a previous token but in this case there has been. I think the better implementation would be to have in `else` branch in the `if` statement in which you set `self.previousLexemeTrailingNewlinePresence` below and in it set `self.previousLexemeTrailingNewlinePresence = .absent`.

If you do that, I think you can also move this entire code block below the `Token text` lexing section and initialize `flags` with `result.flags` and avoid the `union` call when forming the result.

Reply:
As you said, it seems like it should have a non-nil value when lexing the previous token.
However, I'm concerned that it would be redundant to assign .absent to each else in the branch that sets previousLexemeTrailingNewlinePresence.

https://github.com/apple/swift-syntax/pull/1661/files#diff-253b01bc981faa185173c1697784a41bf1b586e91ec7b4de806dfd2646db52c3R1900

The code above makes it look like I need to put previousLexemeTrailingNewlinePresence = .absent in the lex logic for each case.
https://github.com/apple/swift-syntax/blob/main/Sources/SwiftParser/Lexer/Cursor.swift#L410-L431

Why not just leave the code where it is, but assign absent as the default after the value is used?
I think we can get some reasonable logic with little modification.

```suggestion
    } else {
      self.previousLexemeTrailingNewlinePresence = .absent
    }
```

Reply:
Additionally, if we move this block of code below the token text lexing section, the value assigned in the token text lexing logic will affect the flag value of the current token, so it would be difficult to move it 🤔

Reply:
Yes, that’s just what I was just about to suggest

Reply:
Similar to `TriviaResult`, I also added a `newlinePresence` property to `Lexer.Result` and changed it so that modifications to the `previousLexemeTrailingNewlinePresence` value can only be made within the `nextToken` method.
The flag initialize logic was also moved under the token text lexing section.